Update Scripts to include Populate Metadata

gusferguson opened the pull request:
Text and screenshots added for the Populate Metadata script.
New feature paragraph added to home page.

Review and proofing needed of scripts.html and index.html

PDFs uploaded to staging.

hflynn commented:
I don't have time to check this now, if everyone is busy on Monday we can always add this after the release.

gusferguson commented:
J-M is quite happy for it to go in a few days after release.

joshmoore commented:
Nice. @eleanorwilliams: any feedback from your side?

jburel commented:
Now that we have re-organised the link. i.e. utility/metadata
Not having it merged, users won't be going to the correct page if they click on a link in the script's header.

gusferguson commented:
@will-moore is reviewing today. 
The page has been changed to scripts.html in a separate PR and that is already merged, so links from alone etc will be good - they will just go to the same old page.

It may be worth just merging this though as long as any issues are minor and I will update/address any issues ASAP.

will-moore commented:
Reading from http://downloads.openmicroscopy.org/help-staging/pdfs/ ...

Screenshots all look great.

At the start "The Populate Metadata script allows the upload of metadata..." maybe say "metadata or analysis results" here in case users are wondering what metadata they might want to use?

I wonder if the 2nd paragraph (and the rest of the intro) could be a much simpler, without getting into details of csv formats etc.
Maybe just a couple of sentences to say that this gets data from a local spread-sheet file with one row per Well into OMERO so that it shows up under "Tables" tab in web when that Well is selected, and show the image there.
Then you can go into csv details etc as Step 1.
I just got confused when starting the 3rd paragraph of intro "After population... " thinking you hadn't told me how to do the population yet!

I think users might get confused with the 2 examples files and "Select the Screen or Plate and attach the metadata .csv file", not being clear that the csv files are different formats and you have to attach it to the right thing. Maybe the whole thing could be described for the simpler Plate scenario. If we really need to include the Screen scenario, we could add that in a final step at the end?

"Detailed guidelines can be seen on the scripts documentation page" links to https://www.openmicroscopy.org/site/support/omero5.2/developers/scripts/ but this doesn't give any more info on Populate Metadata or the csv format.

Minor: populateMetadata_example_Screen.csv and populateMetadata_example_Plate.csv are all lowercase plate & screen names in the downloads folder.

Step 1. ...attach the metadata... The 2nd sentence "If there is.." is a little confusing as it seems to be more details on how to attach the .csv (rather than being part of the next Step of running the script). 
Maybe make the first sentence a bit longer with the csv details above or add "...using the [+] button under the Attachments tab".
Then you could add an extra step for "Select the Plate and the csv file: If there is more..."

In the Note: maybe replace PyTable with "OMERO.table file" since users won't know this is a PyTable?

sbesson commented on scripts.html:
```diff
@@ -4,16 +4,219 @@
-description: Scripts allow users to add server-side functionality to the OMERO clients.
+description: Scripts allow additional server-side functionality to be added by users to the OMERO clients.
 version:
     - yes
```
Can we add a redirect from the removed utility scripts page by adding:

```
redirect_from:
  - /utility-scripts.html
```

to the header?

gusferguson replied:
No - I don't think you can the way this is set up with Jekyll.

gusferguson replied:
I think it has to be set up on the server.

sbesson replied:
@gusferguson: the `redirect-from` is the way we mitigated the SLS renaming impact i.e. it redirects http://help.openmicroscopy.org/cls-public-demo.html to http://help.openmicroscopy.org/sls-public-demo.html - see https://github.com/openmicroscopy/ome-help/pull/119.

gusferguson replied:
OK thanks - I will look at that. Github email notification failures not making life easy!

sbesson replied:
:+1: As discussed, happy to work on documenting this feature either in the README or in any other convenient place once this epic PR is merged.

gusferguson replied:
No problem - I will do it - a few other things to update as well.

gusferguson replied:
https://github.com/openmicroscopy/ome-help/pull/153

gusferguson commented:
@will-moore - thanks for the comments and feedback.

I think they do make it clearer.

I left this in as @jburel indicated there would be some more detail in there (still to be added I presume):

>"Detailed guidelines can be seen on the scripts documentation page" links to https://www.openmicroscopy.org/site/support/omero5.2/developers/scripts/ but this doesn't give any more info on Populate Metadata or the csv format.

All in all I think it may reduce confusion to get this merged ASAP and then sort out redirects and other details.

NB: I will need a heads-up before this goes live so that I can update the 5.2.2 archive on downloads staging (still has Utility Scripts in it).

eleanorwilliams commented:
Here are my comments:

1. First paragraph under 'Using Scripts'. It wasn't clear to me where the Populate Metadata script fitted with the image of 'analysis scripts, export scripts, figure scripts, import scripts, util scripts'.  So maybe change the second sentence to be 'to add functionality  such as the Populate Metadata "import script" and "utility scripts".

2. First sentence under 'Populate Metadata Script'.  I agree with Will that it might need to be made more clear what type of metadata we are talking about.  You could be very explicit and change the sentence to be "The Populate Metadata script allows the upload of metadata associated with High Content Screen (HCS) data such reagents (e.g. siRNA or compound) applied to each well, the gene targeted, or results associated with reagents such as phenotypes observed, or scores."

3. Third sentence under 'Populate Metadata Script'. Its not clear what is meant by "the data in the .csv must be formatted appropriately" and it sounds a bit scary.  Really the only difference is whether the Plate column is included or not so maybe you could change the text to something like "If metadata is to be added to several or all plates in a Screen then a column with the Plate name should be included in the .csv and the metadata will be attached to the Screen.  If metadata is to be added to a single plate, then the Plate name column can be omitted and the metadata will be attached to the Plate specified when the script is run."

4. Third paragraph under 'Populate Metadata Script'.  I also thought it suddently jumps to 'After population' without saying how to populate.  Could say "The .csv file is attached to the Screen or Plate, then the Populate Metadata script is run, as show in detail in the steps below.  After population ..."

5. "Note" after step 5.  Change the last sentence to be "It must not be removed or deleted unless you want to remove all the annotations.".  People might want to update the annotations to add results, or if they have made a mistake in the .csv file etc, so would be good to know how to get rid of it.

jburel commented:
https://www.openmicroscopy.org/site/support/omero5.2/developers/scripts/  the page will be reviewed when @hflynn gets back. Comment added to card on release board and already in the pipeline.

will-moore commented:
@gusferguson It seems most of my comments aren't addressed by your commit?

gusferguson commented:
Built on http://help.staging.openmicroscopy.org/scripts.html

will-moore commented:
@gusferguson Ah, OK that looks a bit better. 
 - You're now missing the "csv -> display in web" figure that I thought was quite useful.
 - I still think that the distinction between Screen and Plate workflow is not clear enough. Either this point needs to be laboured (as @eleanorwilliams suggests) or just remove the Screen workflow entirely as I suggested above (and describe it at the end if needed). With "Screen or Plate" everywhere in the text it might confuse people as to which scenario is which.

jburel commented:
If we need some extra work e.g. screenworkflow, this can be done in a follow-up PR
suggestion: tackle in this PR remaining minor edits and the redirect so we can go live. (currently links in script headers are broken w/o this PR in)

gusferguson commented:
@eleanorwilliams - thanks for the comments.

Have added redirect.

I have added what you suggested, but:
-  due to conflict with Will's desire to keep the intro successful I added your Point 3 under the "Populating a Screen with Metadata or Analysis Results" paragraph after the steps which I have restricted to adding to a Plate (dropping mention of screen) i.e. a bit of a compromise
- Point 4 superseded by other changes

This should cover all points (more or less).

PDFs and archive updated on downloads staging.

As suggested by @jburel - it is probably best to get this merged and then thrash out any other issues/fine tuning with a subsequent PR otherwise the Help website is not updated to 5.2.2.

Rebuilding Help staging as soon as Travis passed.

gusferguson commented:
Erk! forgot to push - doing now.

gusferguson commented:
Built on staging.

eleanorwilliams commented:
@gusferguson.  Looks good to me now.

jburel commented:
redirect works too.

jburel commented:
Thanks all merging
